src/com/xilinx/rapidwright/rwroute/RouteNodeGraph.java

```diff
@@ -114,20 +116,28 @@ public class RouteNodeGraph {
     /** Flag for whether design targets the Versal series */
     protected final boolean isVersal;
 
-    public RouteNodeGraph(Design design, RWRouteConfig config) {
-        this(design, config, new HashMap<>());
+    protected final Map<TileTypeEnum, Integer> baseWireCounts;
+
+    protected final static int MAX_OCCUPANCY = 256;
+    protected final float[] presentCongestionCosts;
+
+    protected static int getTileCount(Design design) {
```

Comment:
Should we add this to the `Device` object?

Reply:
Yes, I think that's a good idea. Specifically, I think we should update the documentation to state that `Tile.getUniqueAddress()` should not exceed a new method that is `Tile.getMaxUniqueAddress()`.

Let's not hold up this PR until the next release though.